docs/DeveloperGuide.md

```diff
@@ -4,9 +4,118 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Classes: overview
```

Comment:
Would it be better to include a table of contents (with links to the relevant sections) for easier navigability?